lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -465,14 +465,7 @@ function dosomething_user_new_user($form, &$form_state) {
   $account = $user;
 
   if (module_exists('dosomething_global')) {
-    // Adjust language based on location
-    $user_country_code = dosomething_settings_get_geo_country_code();
-    $language = dosomething_global_convert_country_to_language($user_country_code);
-    $system_languages = language_list();
-    if (!in_array($langauge, $system_languages)) {
-      $language = 'en-global';
-    }
-    user_save($account, ['language' => $language]);
+    dosomething_user_set_global_attributes();
```

Comment:
instead of calling this from here, let's call it in the form_alter, like `dosomething_user_new_user()` is called